pkg/controller/bastion/resourcesdefine.go

```diff
@@ -0,0 +1,161 @@
+// Copyright (c) 2021 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bastion
+
+import (
+	"encoding/base64"
+
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2021-03-01/compute"
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2020-05-01/network"
+	"github.com/Azure/go-autorest/autorest/to"
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+	gardenerutils "github.com/gardener/gardener/pkg/utils"
+)
+
+func nicDefine(opt *Options, ip *network.PublicIPAddress, subnet *network.Subnet) *network.Interface {
+	return &network.Interface{
+		Name:     &opt.NicName,
+		Location: &opt.Location,
+		InterfacePropertiesFormat: &network.InterfacePropertiesFormat{
+			IPConfigurations: &[]network.InterfaceIPConfiguration{
+				{
+					Name: to.StringPtr("ipConfig1"),
+					InterfaceIPConfigurationPropertiesFormat: &network.InterfaceIPConfigurationPropertiesFormat{
+						Subnet: &network.Subnet{
+							ID: subnet.ID},
+						PrivateIPAllocationMethod: network.Dynamic,
+						PublicIPAddress:           ip,
+					},
+				},
+			},
+		},
+		Tags: opt.Tags,
+	}
+}
+
+func publicIPAddressDefine(opt *Options) *network.PublicIPAddress {
+	return &network.PublicIPAddress{
+		Name:     &opt.BastionPublicIPName,
+		Location: &opt.Location,
+		Sku: &network.PublicIPAddressSku{
+			Name: network.PublicIPAddressSkuNameStandard,
+		},
+		PublicIPAddressPropertiesFormat: &network.PublicIPAddressPropertiesFormat{
+			PublicIPAddressVersion:   network.IPv4,
+			PublicIPAllocationMethod: network.Static,
+		},
+		Tags: opt.Tags,
+	}
+}
+
+func computeInstanceDefine(opt *Options, bastion *extensionsv1alpha1.Bastion, adminPassword string) *compute.VirtualMachine {
+	return &compute.VirtualMachine{
+		Location: &opt.Location,
+		VirtualMachineProperties: &compute.VirtualMachineProperties{
+			HardwareProfile: &compute.HardwareProfile{
+				VMSize: compute.VirtualMachineSizeTypesStandardB1s,
```

Comment:
Maybe we fetch this information in a later step from the respective CloudProfile instead of hardcode it here.
Same for the Image settings, but as said this not in scope for this PR.